gnome-image-installer/pages/finished/gis-finished-page.c

```diff
@@ -206,16 +206,43 @@ gis_finished_page_shown (GisPage *page)
   if (error != NULL)
     {
       GisAssistant *assistant = gis_driver_get_assistant (page->driver);
+      const gchar *heading = NULL;
+      const gchar *detail = error->message;
 
       if (error->domain == GIS_UNATTENDED_ERROR)
-        gtk_label_set_text (priv->error_heading_label,
-                            _("Oops, something is wrong with your unattended installation configuration."));
-      /* Otherwise, leave the default message (indicating a problem with the
-       * image file). TODO: handle other domains that indicate problems
-       * elsewhere.
-       */
-
-      gtk_label_set_text (priv->error_label, error->message);
+        {
+          heading = _("Oops, something is wrong with your unattended installation configuration.");
+        }
+      else if (error->domain == GIS_IMAGE_ERROR)
+        {
+          heading = _("Oops, something is wrong with your Endless OS file.");
+        }
+      else if (error->domain == GIS_DISK_ERROR)
+        {
+          heading = _("Oops, something went wrong while finding suitable disks to reformat.");
+        }
+      else if (error->domain == GIS_INSTALL_ERROR)
+        {
+          heading = _("Oops, something went wrong while reformatting.");
+        }
+      else if (g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+        {
+          heading = error->message;
```

Comment:
I assume you’re OK with this (most likely) being `_("Operation was cancelled")`? There could potentially be issues here if someone does `g_set_error (error, G_IO_ERROR, G_IO_ERROR_CANCELLED, "some untranslated other message")`, but I think that’s unlikely.

Reply:
Yes, I think that's fine. (Actually, at present the message can only be `_("Unattended installation was cancelled.")`; there are no GCancellables involved.)